crypto/crypto/src/x25519.rs

```diff
@@ -93,13 +96,56 @@ impl PrivateKey {
         let shared_secret = self.0.diffie_hellman(&remote_public_key);
         shared_secret.as_bytes().to_owned()
     }
+
+    /// Deserialize an X25119 PrivateKey given a canonical representation as an
+    /// bytes, following the XEdDSA approach.
+    ///
+    /// This will FAIL if the passed-in byte representation converts to a
+    /// non-canonical scalar in the X25519 sense (and thus cannot correspond to
+    /// a X25519 valid key without bit-mangling).
+    ///
+    /// This is meant to compensate for the poor key storage capabilities of
+    /// key management solutions, and NOT to promote double usage of keys under
+    /// several schemes, which would lead to BAD vulnerabilities.
+    pub fn from_scalar_bytes(scalar_slice: &[u8]) -> Result<Self, CryptoMaterialError> {
```

Comment:
Why is the argument called scalar_slice? Is this function simply generates an x25519 private key from a random byte array?

Reply:
The function will return `Err` on precisely those inputs whose ms-half-of-sha512 image are not a canonical X25519 Scalar.

Reply:
Don't we want instead a function ed25519 sk -> x25519 sk?
Where the ed25519 sk is an UN-expanded secret key (i.e. pre SHA512)?

Reply:
I was just confused by the naming of the function's argument. Makes sense now!